internal/api/health.go

```diff
@@ -1,12 +1,42 @@
 package api
 
 import (
+	"errors"
+	"fmt"
+	"github.com/openHPI/poseidon/internal/config"
+	"github.com/openHPI/poseidon/internal/environment"
+	"github.com/openHPI/poseidon/pkg/dto"
 	"net/http"
 )
 
+var ErrorPrewarmingPoolDepleting = errors.New("the prewarming pool is depleting")
+
 // Health handles the health route.
 // It responds that the server is alive.
 // If it is not, the response won't reach the client.
-func Health(writer http.ResponseWriter, _ *http.Request) {
-	writer.WriteHeader(http.StatusNoContent)
+func Health(manager environment.Manager) http.HandlerFunc {
+	return func(writer http.ResponseWriter, request *http.Request) {
+		if err := checkPrewarmingPool(manager); err != nil {
+			writeInternalServerError(writer, err, dto.ErrorUnknown, request.Context())
+			return
+		}
+
+		writer.WriteHeader(http.StatusNoContent)
+	}
+}
+
+func checkPrewarmingPool(manager environment.Manager) error {
+	prewarmingPoolOk := true
+	err := ErrorPrewarmingPoolDepleting
+	for _, data := range manager.Statistics() {
+		if float64(data.IdleRunners)/float64(data.PrewarmingPoolSize) < config.Config.Server.Alert.PrewarmingPoolThreshold {
+			prewarmingPoolOk = false
+			err = fmt.Errorf("%w: environment %d", err, data.ID)
```

Comment:
This can lead to some interesting error messages:

```
    "message": "the prewarming pool is depleting: environment 29: environment 34",
```

Here, I would have expected that only one colon (`:`) is included and the second environment is added with a comma (`,`).